src/SWP/Bundle/ContentBundle/Model/Article.php

```diff
@@ -446,4 +481,71 @@ public function setMetadata(array $metadata): void
     {
         $this->metadata = $metadata;
     }
+
+    public function addTextExtra(ArticleExtraTextFieldInterface $articleExtra): void
+    {
+        if (!$this->extraTextFields->contains($articleExtra)) {
+            $this->extraTextFields[$articleExtra->getFieldName()] = $articleExtra;
+            $articleExtra->setArticle($this);
+        }
+    }
+
+    public function addEmbedExtra(ArticleExtraEmbedFieldInterface $articleExtra): void
+    {
+        if (!$this->extraEmbedFields->contains($articleExtra)) {
+            $this->extraEmbedFields[$articleExtra->getFieldName()] = $articleExtra;
+            $articleExtra->setArticle($this);
+        }
+    }
+
+    public function removeExtraTextFields(ArticleExtraFieldInterface $articleExtra): void
+    {
+        $this->extraTextFields->removeElement($articleExtra);
+    }
+
+    public function removeExtraEmbedFields(ArticleExtraEmbedFieldInterface $articleExtra): void
+    {
+        $this->extraEmbedFields->removeElement($articleExtra);
+    }
+
+    public function getExtraTextFields(): Collection
+    {
+        return $this->extraTextFields;
+    }
+
+    public function getExtraEmbedFields(): Collection
+    {
+        return $this->extraEmbedFields;
+    }
+
+    public function setExtraFields(array $extra): void
+    {
+        if (0 === count($extra)) {
+            return;
+        }
+
+        foreach ($this->getExtraTextFields() as $extra) {
+            $this->removeExtraTextFields($extra);
+        }
+
+        foreach ($this->getExtraEmbedFields() as $extra) {
+            $this->removeExtraEmbedFields($extra);
+        }
+
+        foreach ($extra as $key => $value) {
+
+            if(is_array($value)) {
+                $embed = new ArticleExtraEmbedField();
```

Comment:
You can create `named constructors` in `ArticleExtraEmbedField` and `ArticleExtraTextField` classes, so this part of the code is more readable, and we will have a one way to construct those objects.